framework/src/modules/chain/blocks/chain.js

```diff
@@ -195,11 +224,29 @@ const saveBlock = async (storage, block, tx) => {
 	parsedBlock.previousBlockId = parsedBlock.previousBlock;
 	delete parsedBlock.previousBlock;
 
-	parsedBlock.transactions.map(transaction => {
+	parsedBlock.transactions.forEach(transaction => {
 		transaction.blockId = parsedBlock.id;
 		return transaction;
 	});
 
+	parsedBlock.transactions = parsedBlock.transactions.map(transaction =>
+		transaction.toJSON()
+	);
+
+	return parsedBlock;
+};
+
+/**
+ * Save block with transactions to database.
+ *
+ * @param {Object} block - Full normalized block
+ * @param {function} cb - Callback function
+ * @returns {Function|afterSave} cb - If SQL transaction was OK - returns safterSave execution, if not returns callback function from params (through setImmediate)
+ * @returns {string} cb.err - Error if occurred
+ */
```

Comment:
These `jsdocs` are outdated. I don't know if it's the scope of this PR but maybe we could update them.